table/get_context.cc

```diff
@@ -226,23 +226,27 @@ bool GetContext::SaveValue(const ParsedInternalKey& parsed_key,
               pinnable_val_->PinSelf(value);
             }
           }
+          if (is_blob_index_) {
+            *is_blob_index_ = (type == kTypeBlobIndex);
+          }
         } else if (kMerge == state_) {
           assert(merge_operator_ != nullptr);
           state_ = kFound;
           if (LIKELY(pinnable_val_ != nullptr)) {
             Status merge_status = MergeHelper::TimedFullMerge(
-                merge_operator_, user_key_, &value,
-                merge_context_->GetOperands(), pinnable_val_->GetSelf(),
+                merge_operator_, user_key_, type, &value,
+                merge_context_->GetOperands(), &type, pinnable_val_->GetSelf(),
                 logger_, statistics_, env_);
             pinnable_val_->PinSelf();
             if (!merge_status.ok()) {
               state_ = kCorrupt;
+            } else if (is_blob_index_ != nullptr) {
+              *is_blob_index_ = (type == kTypeBlobIndex);
```

Comment:
should update `state_` too ?

Reply:
I think state_ is kFound here. kBlobIndex is an abnormal state.